examples/ui-demo/src/components/small-cards/SolanaNftCard.tsx

```diff
@@ -0,0 +1,226 @@
+import { useToast } from "@/hooks/useToast";
+import { useSolanaTransaction } from "@account-kit/react";
+import {
+  Keypair,
+  PublicKey,
+  SystemProgram,
+  VersionedTransaction,
+} from "@solana/web3.js";
+
+import {
+  ExtensionType,
+  LENGTH_SIZE,
+  TOKEN_2022_PROGRAM_ID,
+  TYPE_SIZE,
+  createInitializeInstruction,
+  createInitializeMetadataPointerInstruction,
+  createInitializeMintInstruction,
+  createUpdateFieldInstruction,
+  getMintLen,
+} from "@solana/spl-token";
+import { pack, TokenMetadata } from "@solana/spl-token-metadata";
+import { LoadingIcon } from "../icons/loading";
+import { Button } from "./Button";
+import { Card } from "./Card";
+import Image from "next/image";
+import { Badge } from "./Badge";
+import { CheckCircleFilledIcon } from "../icons/check-circle-filled";
+import { useState } from "react";
+
+type TransactionState = "idle" | "signing" | "sponsoring" | "complete";
+
+export const SolanaNftCard = () => {
+  const { setToast } = useToast();
+  const {
+    sendTransactionAsync,
+    isPending,
+    connection,
+    signer: solanaSigner,
+  } = useSolanaTransaction();
+  const [transactionState, setTransactionState] =
+    useState<TransactionState>("idle");
+
+  const handleCollectNFT = async () => {
+    try {
+      if (!solanaSigner) throw new Error("No signer found");
+      if (!connection) throw new Error("No connection found");
+      setTransactionState("signing");
+      setTransactionState("sponsoring");
```

Comment:
seems weird to me that we go from `signing` to `sponsoring` w/o anything happening in-between?

Reply:
nvm, i see @blakecduncan already commented this on #1578 .

Reply:
Yeah, I pinged with margaret about this earlier, I agree.